marketplace-service/src/main/java/com/axonivy/market/service/impl/ProductServiceImpl.java

```diff
@@ -574,4 +562,70 @@ public void transferComputedDataFromDB(Product product) {
     );
   }
 
+  @Override
+  public Product renewProductById(String id) {
+    Product product = new Product();
+    productRepository.findById(id).ifPresent(foundProduct -> {
+          ProductFactory.transferComputedPersistedDataToProduct(foundProduct, product);
```

Comment:
When you renew the productId, you remove the image, productModuleContent, productJsonContent, and product, then you sync the product again.

This can affect users who are using this product on the website UI or designer.

I suggest enabling the transaction of MongoDB and putting all features in the transaction. This will ensure that it does not affect users or lose data when a feature crashes.